tree/tree/src/TBranch.cxx

```diff
@@ -2592,51 +2596,59 @@ Int_t TBranch::WriteBasket(TBasket* basket, Int_t where)
       fEntryOffsetLen = 2*nevbuf; // assume some fluctuations.
    }
 
-   Int_t nout  = basket->WriteBuffer();    //  Write buffer
-   fBasketBytes[where]  = basket->GetNbytes();
-   fBasketSeek[where]   = basket->GetSeekKey();
-   Int_t addbytes = basket->GetObjlen() + basket->GetKeylen();
-   TBasket *reusebasket = 0;
-   if (nout>0) {
-      // The Basket was written so we can now safely reuse it.
-      fBaskets[where] = 0;
+   auto do_updates = [=]() {
```

Comment:
This lambda is added to the `task_group` and `WriteBasket` returns.  Hence, the reference becomes invalid.  You must actually copy the pointer (as the pointed-to object's lifetime is guaranteed to be longer than the lambda), but not reference to the pointer itself.